gems/aws-sdk-core/lib/aws-sdk-core/rest/request/querystring_builder.rb

```diff
@@ -71,6 +62,53 @@ def timestamp(ref, value)
           end
         end
 
+        def generate_query(ref, value)
+          case ref.shape
+          when TimestampShape
+            value = timestamp(ref, value)
+          when *SUPPORTED_TYPES
+            value = value.to_s
+          else
+            raise NotImplementedError
+          end
+          "#{ref.location_name}=#{escape(value)}"
+        end
+
+        def generate_query_list(ref, values)
+          case ref.shape.member.shape
+          when TimestampShape
+            list_of_timestamps(ref, values)
```

Comment:
I think rather than having a specific method for timestamps, you can use the generic `generate_query`.

I think if you do:
```
values.map { |value| generate_query(ref.shape.member, value) }
```

Reply:
Edit: NVM, this doesn't quite work since the current `generate_query` does rely on the location_name.

As a different suggestion, how about a generic `query_value` method which takes the shape and value and returns just the correctly serialized value, eg:

```
def query_value(ref, value)
  case ref.shape
  when TimestampShape
     escape(timestamp(ref, value))
   when *SUPPORTED_TYPES
      escape(value.to_s)
    else
      raise NotImplementedError
     end
end
```

And then both single values and lists can use it I think.

Reply:
Looks so much cleaner, thanks :)